benchmark/jsonresultset/jsonresultset_test.go

```diff
@@ -0,0 +1,130 @@
+// This code is to profile a large json result set query. It is basically similar to selectmany example code but
+// leverages benchmark framework.
+package jsonresultset
+
+import (
+	"flag"
+	"log"
+	_ "net/http/pprof"
+	"os"
+	"testing"
+
+	"database/sql"
+
+	"context"
+	"os/signal"
+
+	"runtime/debug"
+
+	"strconv"
+	"strings"
+
+	sf "github.com/snowflakedb/gosnowflake"
+)
+
+func TestJsonResultSet(t *testing.T) {
+	runJSONResultSet()
+}
+
+func BenchmarkJsonResultSet(*testing.B) {
+	runJSONResultSet()
+}
+
+// getDSN constructs a DSN based on the test connection parameters
+func getDSN() (dsn string, cfg *sf.Config, err error) {
+	env := func(k string, failOnMissing bool) string {
+		if value := os.Getenv(k); value != "" {
+			return value
+		}
+		if failOnMissing {
+			log.Fatalf("%v environment variable is not set.", k)
+		}
+		return ""
+	}
+
+	account := env("SNOWFLAKE_TEST_ACCOUNT", true)
+	user := env("SNOWFLAKE_TEST_USER", true)
+	password := env("SNOWFLAKE_TEST_PASSWORD", true)
+	host := env("SNOWFLAKE_TEST_HOST", false)
+	port := env("SNOWFLAKE_TEST_PORT", false)
+	protocol := env("SNOWFLAKE_TEST_PROTOCOL", false)
+	role := env("SNOWFLAKE_TEST_ROLE", false)
+
+	// Use the customer JSON Decoder
+	s := env("SNOWFLAKE_TEST_CUSTOME_JSON_DECODER_ENABLED", true)
+	sf.CustomJSONDecoderEnabled = strings.EqualFold("true", s)
+
+	// Set the maximum chunk download workers
+	n := env("SNOWFLAKE_TEST_MAX_CHUNK_DOWNLOAD_WORKERS", false)
```

Comment:
Change the number of workers in the test. Changing this help to mitigate connection timeout.